common/src/node/cosmos_sdk.rs

```diff
@@ -287,6 +290,50 @@ pub async fn broadcast_tx_sync(
         .map_err(RestError::RequestError)
 }
 
+/// The choice for Tendermint JSON-RPC transaction broadcast endpoint
+pub enum TxBroadcastMode {
+    /// returns the checkTx result
+    Sync,
+    /// returns immediately
+    Async,
+    /// returns the checkTx + deliverTx results or times out
+    /// (mainly for development)
+    Commit,
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+macro_rules! broadcast_tx {
+    ($mode:ident, $raw_signed_tx:expr, $tendermint_rpc_url:expr) => {{
+        let request = request::Wrapper::new($mode::Request {
+            tx: $raw_signed_tx.into(),
+        });
+        let rpc_result = reqwest::blocking::Client::new()
+            .post($tendermint_rpc_url)
+            .json(&request)
+            .send()
+            .map_err(RestError::RequestError)?
+            .json::<response::Wrapper<$mode::Response>>()
+            .map_err(RestError::RequestError)?
+            .into_result()
+            .map_err(|_e| RestError::MissingResult)?;
+
+        Ok(rpc_result.into())
+    }};
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn broadcast_tx_blocking(
```

Comment:
not familiar with tx broadcast mode, just a little bit confuse/curious why the function name is `blocking` but it also has `async` mode?

Reply:
I guess if it simply three functions, broadcast_tx_sync_blocking, broadcast_tx_commit_blocking, broadcast_tx_async_blocking...

Reply:
I know why now, broadcast_tx_blocking and broadcast_tx, both add a new mode parameter.

Reply:
a bit confusing I guess: it's a mode on the server-side endpoint... so even though it's broadcasted "async", the client execution is still blocking, but the endpoint returns immediately... as you said, it'd be three functions if they all went as separate bindings